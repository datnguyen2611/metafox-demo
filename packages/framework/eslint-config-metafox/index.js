const restrictedGlobals = require('confusing-browser-globals');

module.exports = {
  root: true,
  // parser: '@babel/eslint-parser',
  parser: '@typescript-eslint/parser',
  extends: ['eslint:recommended'],
  plugins: [
    'import',
    'flowtype',
    'jsx-a11y',
    'react',
    'react-hooks',
    'jest-dom'
  ],

  env: {
    browser: true,
    commonjs: true,
    es6: true,
    jest: true,
    node: true
  },
  parserOptions: {
    ecmaVersion: 2020,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true
    }
  },

  settings: {
    react: {
      version: 'detect'
    }
  },
  overrides: [
    {
      files: ['**/*.ts?(x)'],
      // parser: '@typescript-eslint/parser',
      parserOptions: {
        ecmaVersion: 2020,
        sourceType: 'module',
        ecmaFeatures: {
          jsx: true
        },
        // typescript-eslint specific options
        warnOnUnsupportedTypeScriptVersion: true
      },
      plugins: ['@typescript-eslint'],
      // If adding a typescript-eslint version of an existing ESLint rule,
      // make sure to disable the ESLint rule here.
      rules: {
        // TypeScript's `noFallthroughCasesInSwitch` option is more robust (#6906)
        'default-case': 'off',
        // 'tsc' already handles this
        // (https://github.com/typescript-eslint/typescript-eslint/issues/291)
        'no-dupe-class-members': 'off',
        // 'tsc' already handles this
        // (https://github.com/typescript-eslint/typescript-eslint/issues/477)
        'no-undef': 'off',

        // Add TypeScript specific rules (and turn off ESLint equivalents)
        '@typescript-eslint/consistent-type-assertions': 'warn',
        'no-array-constructor': 'off',
        '@typescript-eslint/no-array-constructor': 'warn',
        'no-use-before-define': 'off',
        '@typescript-eslint/no-use-before-define': [
          'error',
          {
            functions: false,
            classes: false,
            variables: false,
            typedefs: false
          }
        ],
        'no-unused-expressions': 'off',
        '@typescript-eslint/no-unused-expressions': [
          'error',
          {
            allowShortCircuit: true,
            allowTernary: true,
            allowTaggedTemplates: true
          }
        ],
        'no-unused-vars': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            args: 'none',
            ignoreRestSiblings: true
          }
        ],
        'no-useless-constructor': 'off',
        '@typescript-eslint/no-useless-constructor': 'warn'
      }
    }
  ],

  // NOTE: When adding rules here, you need to make sure they are compatible with
  // `typescript-eslint`, as some rules such as `no-array-constructor` aren't compatible.
  rules: {
    // http://eslint.org/docs/rules/
    'array-callback-return': 'warn',
    'default-case': [
      'warn',
      {
        commentPattern: '^no default$'
      }
    ],
    eqeqeq: ['warn', 'smart'],
    'new-parens': 'warn',
    'no-array-constructor': 'warn',
    'no-caller': 'warn',
    'no-cond-assign': ['warn', 'except-parens'],
    'no-const-assign': 'error',
    'no-control-regex': 'warn',
    'no-delete-var': 'warn',
    'no-dupe-args': 'warn',
    'no-dupe-class-members': 'error',
    'no-duplicate-imports': [
      2,
      {
        includeExports: false
      }
    ],
    'no-dupe-keys': 'warn',
    'no-duplicate-case': 'warn',
    'no-empty-character-class': 'warn',
    'no-empty-pattern': 'warn',
    'no-eval': 'warn',
    'no-ex-assign': 'warn',
    'no-extend-native': 'warn',
    'no-extra-bind': 'warn',
    'no-extra-label': 'warn',
    'no-fallthrough': 'warn',
    'no-func-assign': 'warn',
    'no-implied-eval': 'warn',
    'no-invalid-regexp': 'warn',
    'no-iterator': 'warn',
    'no-label-var': 'warn',
    'no-labels': [
      'warn',
      {
        allowLoop: true,
        allowSwitch: false
      }
    ],
    'no-lone-blocks': 'warn',
    'no-loop-func': 'warn',
    'no-mixed-operators': [
      'warn',
      {
        groups: [
          ['&', '|', '^', '~', '<<', '>>', '>>>'],
          ['==', '!=', '===', '!==', '>', '>=', '<', '<='],
          ['&&', '||'],
          ['in', 'instanceof']
        ],
        allowSamePrecedence: false
      }
    ],
    'no-multi-str': 'warn',
    'no-native-reassign': 'warn',
    'no-negated-in-lhs': 'warn',
    'no-new-func': 'warn',
    'no-new-object': 'warn',
    'no-new-symbol': 'warn',
    'no-new-wrappers': 'warn',
    'no-obj-calls': 'warn',
    'no-octal': 'warn',
    'no-octal-escape': 'warn',
    'no-redeclare': [
      'warn',
      {
        builtinGlobals: false
      }
    ],
    'no-regex-spaces': 'warn',
    'no-restricted-syntax': ['warn', 'WithStatement'],
    'no-script-url': 'warn',
    'no-self-assign': 'warn',
    'no-self-compare': 'warn',
    'no-sequences': 'warn',
    'no-shadow-restricted-names': 'warn',
    'no-sparse-arrays': 'warn',
    'no-template-curly-in-string': 'warn',
    'no-this-before-super': 'warn',
    'no-throw-literal': 'warn',
    'no-undef': 'error',
    'no-restricted-globals': ['error'].concat(restrictedGlobals),
    'no-unreachable': 'warn',
    'no-unused-expressions': [
      'error',
      {
        allowShortCircuit: true,
        allowTernary: true,
        allowTaggedTemplates: true
      }
    ],
    'no-unused-labels': 'warn',
    'no-unused-vars': [
      'warn',
      {
        args: 'none',
        ignoreRestSiblings: true
      }
    ],
    'no-use-before-define': [
      'warn',
      {
        functions: false,
        classes: false,
        variables: false
      }
    ],
    'no-useless-computed-key': 'warn',
    'no-useless-concat': 'warn',
    'no-useless-constructor': 'warn',
    'no-useless-escape': 'warn',
    'no-useless-rename': [
      'warn',
      {
        ignoreDestructuring: false,
        ignoreImport: false,
        ignoreExport: false
      }
    ],
    'no-with': 'warn',
    'space-before-function-paren': [
      2,
      {
        anonymous: 'always',
        named: 'never',
        asyncArrow: 'always'
      }
    ],
    'template-curly-spacing': [2, 'never'],
    'nonblock-statement-body-position': 0,
    'no-whitespace-before-property': 'warn',
    'react-hooks/exhaustive-deps': 'warn',
    'require-yield': 'warn',
    'yield-star-spacing': [1, 'after'],
    'rest-spread-spacing': [1, 'never'],
    strict: ['warn', 'never'],
    'unicode-bom': ['warn', 'never'],
    'use-isnan': 'warn',
    'wrap-regex': 'off',
    'sort-imports': [
      0,
      {
        ignoreCase: false,
        ignoreDeclarationSort: false,
        ignoreMemberSort: true,
        memberSyntaxSortOrder: ['none', 'all', 'single', 'multiple'],
        allowSeparatedGroups: false
      }
    ],
    'valid-typeof': 'warn',
    'no-restricted-properties': [
      'error',
      {
        object: 'require',
        property: 'ensure',
        message:
          'More info: https://facebook.github.io/create-react-app/docs/code-splitting'
      },
      {
        object: 'System',
        property: 'import',
        message:
          'More info: https://facebook.github.io/create-react-app/docs/code-splitting'
      }
    ],
    'no-multi-spaces': [
      2,
      {
        ignoreEOLComments: true,
        exceptions: {
          Property: true
        }
      }
    ],
    'comma-spacing': [
      2,
      {
        before: false,
        after: true
      }
    ],
    'getter-return': 'warn',
    'import/first': 'error',
    'import/no-amd': 'error',
    'import/no-anonymous-default-export': 'warn',
    'import/no-webpack-loader-syntax': 'error',
    'react/forbid-foreign-prop-types': [
      'warn',
      {
        allowInPropTypes: true
      }
    ],
    'react/jsx-one-expression-per-line': 0,
    'react/jsx-no-literals': 'off',
    'react/jsx-props-no-multi-spaces': 'warn',
    'react/jsx-no-useless-fragment': 'warn',
    'react/jsx-no-script-url': 'error',
    'react/jsx-no-target-blank': 'warn',
    'react/jsx-no-comment-textnodes': 'warn',
    'react/jsx-no-duplicate-props': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-sort-props': [
      0,
      {
        callbacksLast: true,
        shorthandFirst: true,
        ignoreCase: true,
        noSortAlphabetically: false,
        reservedFirst: true
      }
    ],
    'react/jsx-pascal-case': [
      'warn',
      {
        allowAllCaps: true,
        ignore: []
      }
    ],
    'react/jsx-wrap-multilines': [
      2,
      {
        declaration: 'parens',
        assignment: 'parens',
        return: 'parens',
        arrow: 'parens',
        condition: 'ignore',
        logical: 'ignore',
        prop: 'ignore'
      }
    ],
    'react/jsx-uses-react': 'warn',
    'react/jsx-uses-vars': 'warn',
    'react/no-danger-with-children': 'warn',
    'react/no-direct-mutation-state': 'warn',
    'react/no-is-mounted': 'warn',
    'react/no-typos': 'error',
    'react/require-render-return': 'error',
    'react/no-string-refs': 'warn',
    'react/no-unsafe': 'warn',
    'react/no-this-in-sfc': 'warn',
    'react/no-unescaped-entities': 'warn',
    'react/no-unknown-property': [
      1,
      {
        ignore: []
      }
    ],
    'react/react-in-jsx-scope': 'error',
    'react/sort-prop-types': 'warn',
    'react/style-prop-object': 'warn',
    'react/forbid-dom-props': 'warn',
    'react/jsx-boolean-value': [1, 'never'],
    'react/jsx-equals-spacing': 'warn',
    'react/jsx-first-prop-new-line': 'warn',
    'react/jsx-child-element-spacing': 'warn',
    'react/jsx-closing-bracket-location': 0,
    'react/jsx-closing-tag-location': 0,
    'react/self-closing-comp': [
      0,
      {
        component: true,
        html: true
      }
    ],
    'react/jsx-max-props-per-line': [
      1,
      {
        maximum: 1,
        when: 'multiline'
      }
    ],
    'react/jsx-fragments': [1, 'syntax'],
    'jsx-a11y/alt-text': 'warn',
    'jsx-a11y/anchor-has-content': 'warn',
    'react/jsx-indent': 0,
    'react/jsx-indent-props': [1, 2],
    'react/jsx-key': 'warn',
    'react/jsx-max-depth': [
      1,
      {
        max: 15
      }
    ],
    'jsx-a11y/anchor-is-valid': [
      'warn',
      {
        aspects: ['noHref', 'invalidHref']
      }
    ],
    'jsx-a11y/aria-activedescendant-has-tabindex': 'warn',
    'jsx-a11y/aria-props': 'warn',
    'jsx-a11y/aria-proptypes': 'warn',
    'jsx-a11y/aria-role': [
      'warn',
      {
        ignoreNonDOM: true
      }
    ],

    // START eslint-recommended
    'no-console': 1,
    'no-constant-condition': 1,
    'no-debugger': 1,
    'no-empty': 1,
    'no-extra-boolean-cast': 1,
    'no-extra-semi': 1,
    'no-inner-declarations': 1,
    'no-irregular-whitespace': 1,
    'no-mixed-spaces-and-tabs': 1,
    // basic others
    indent: 0,
    'max-len': [
      2,
      120,
      4,
      {
        ignoreComments: true,
        ignoreUrls: true
      }
    ],
    'no-tabs': [
      1,
      {
        allowIndentationTabs: false
      }
    ],
    quotes: [2, 'single', { avoidEscape: true }],
    semi: [2, 'always'],
    'no-multiple-empty-lines': [
      2,
      {
        max: 1
      }
    ],
    'comma-dangle': [
      2,
      {
        arrays: 'never',
        objects: 'never',
        imports: 'never',
        exports: 'never',
        functions: 'never'
      }
    ],
    'padding-line-between-statements': [
      2,
      { blankLine: 'always', prev: 'if', next: '*' },
      { blankLine: 'always', prev: 'directive', next: '*' },
      { blankLine: 'always', prev: '*', next: ['return', 'if', 'function'] },
      { blankLine: 'always', prev: 'block-like', next: 'const' },
      {
        blankLine: 'always',
        prev: 'const',
        next: ['block-like', 'multiline-expression']
      },
      { blankLine: 'never', prev: 'import', next: 'import' },
      // { blankLine: 'never', prev: 'export', next: 'export' },
      {
        blankLine: 'always',
        prev: 'import',
        next: ['const', 'function', 'export', 'class']
      },
      { blankLine: 'always', prev: 'block', next: 'multiline-const' }
    ],
    'prefer-arrow-callback': [2],
    'object-shorthand': [2],
    'dot-location': [1, 'property'],
    'one-var': [2, 'never'],
    'no-var': [2],
    'prefer-template': [1],
    'prefer-const': ['error'],
    'prefer-destructuring': [0],
    'no-bitwise': [0],
    'id-length': [
      'error',
      {
        properties: 'never',
        exceptions: [
          'x',
          'y',
          'q',
          'i',
          'I',
          'e',
          'u',
          'n',
          'k',
          'a',
          'b',
          'c',
          's',
          't',
          '_',
          'v',
          'w',
          'h'
        ]
      }
    ],
    'func-names': [1, 'always'],
    'no-confusing-arrow': [
      1,
      {
        allowParens: true
      }
    ],
    yoda: 0,
    'object-curly-spacing': [2, 'always'],
    'array-bracket-spacing': 0,
    'no-class-assign': 'error',
    'space-unary-ops': [
      2,
      {
        words: false,
        nonwords: false
      }
    ],
    'switch-colon-spacing': [
      2,
      {
        after: true,
        before: false
      }
    ],
    'keyword-spacing': [
      'error',
      {
        after: true
      }
    ],
    'space-infix-ops': [
      2,
      {
        int32Hint: false
      }
    ],
    'space-before-blocks': [2, 'always'],
    'space-in-parens': [2, 'never'],
    'spaced-comment': [2, 'always', { markers: ['/'] }],
    'arrow-spacing': [
      2,
      {
        before: true,
        after: true
      }
    ],
    'arrow-parens': 0,
    'linebreak-style': [1, 'unix'],
    'operator-linebreak': 0,
    'function-paren-newline': 0,
    'react/prefer-stateless-function': 'off',
    'import/newline-after-import': [0],
    'react/jsx-filename-extension': [0],
    'react/forbid-prop-types': [0],
    'import/no-named-as-default': [0],
    'import/no-named-as-default-member': [0],
    'import/no-extraneous-dependencies': [0],
    'import/no-unresolved': [0],
    'import/extensions': [0],
    // END eslint-recommended
    'jsx-a11y/aria-unsupported-elements': 'warn',
    'jsx-a11y/heading-has-content': 'warn',
    'jsx-a11y/iframe-has-title': 'warn',
    'jsx-a11y/img-redundant-alt': 'warn',
    'jsx-a11y/no-access-key': 'warn',
    'jsx-a11y/no-distracting-elements': 'warn',
    'jsx-a11y/no-redundant-roles': 'warn',
    'jsx-a11y/role-has-required-aria-props': 'warn',
    'jsx-a11y/role-supports-aria-props': 'warn',
    'jsx-a11y/scope': 'warn',
    'react-hooks/rules-of-hooks': 'error',
    'flowtype/define-flow-type': 'warn',
    'flowtype/require-valid-file-annotation': 'warn',
    'flowtype/use-flow-type': 'warn',

    // eslint-plugin-jest-dom
    // https://github.com/testing-library/eslint-plugin-jest-dom
    'jest-dom/prefer-checked': 'error',
    'jest-dom/prefer-enabled-disabled': 'error',
    'jest-dom/prefer-required': 'error',
    'jest-dom/prefer-to-have-attribute': 'error'
  }
};
