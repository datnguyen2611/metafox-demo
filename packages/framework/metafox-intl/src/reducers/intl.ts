/**
 * @type: reducer
 * name: intl
 */
import { createAction, createReducer } from '@reduxjs/toolkit';
import { AppState } from '@metafox/intl/types';

const setMessages = createAction('intl/setMessages');

export default createReducer<AppState>(
  {
    messages: {}
  },
  builder => {
    builder.addCase(setMessages, (state, action) => {
      state.messages = action.payload;
    });
  }
);
