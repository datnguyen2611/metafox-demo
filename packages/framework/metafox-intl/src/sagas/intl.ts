/**
 * @type: saga
 * name: metafox.intl
 */

import {
  APP_BOOTSTRAP,
  getGlobalContext,
  GlobalState,
  LOCALE_GROUP,
  IS_ADMINCP,
  USE_BOOTSTRAP_CACHE as CACHED
} from '@metafox/framework';
import { get } from 'lodash';
import { put, select, takeLeading } from 'redux-saga/effects';

const CACHED_PHRASE_KEY = IS_ADMINCP ? 'phrases.admin' : 'phrases';

export function* bootstrapIntl() {
  // load locale data/

  const { apiClient, localStore, preferenceBackend } =
    yield* getGlobalContext();

  try {
    let data: Record<string, string> = CACHED
      ? localStore.getJSON(CACHED_PHRASE_KEY)
      : undefined;

    const revision = CACHED ? get(data, 'revision', 'now') : 'now';

    const defaultMessages = yield select(
      (state: GlobalState) => state.intl.messages
    );

    const response = yield apiClient.get(
      `core/translation/${LOCALE_GROUP}/auto/${revision}`
    );

    const keepCached = get(response, 'data.data.keepCached', false);
    const $locale = get(response, 'data.data.$locale');

    if ($locale) {
      preferenceBackend.set('userLanguage', $locale);
      // handle locale changed.
    }

    if (!keepCached) {
      data = response.data.data;
      localStore.set(CACHED_PHRASE_KEY, JSON.stringify(data));
    }

    yield put({
      type: 'intl/setMessages',
      payload: { ...defaultMessages, ...data }
    });
    yield put({ type: '@bootstrap/intl/DONE' });
  } catch (err) {
    yield put({ type: '@bootstrap/intl/DONE', error: true });
  }
}

const sagas = [takeLeading(APP_BOOTSTRAP, bootstrapIntl)];

export default sagas;
