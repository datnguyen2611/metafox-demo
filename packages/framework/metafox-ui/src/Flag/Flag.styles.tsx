import { Theme } from '@mui/material';
import { createStyles, makeStyles } from '@mui/styles';

export default makeStyles(
  (theme: Theme) =>
    createStyles({
      root: {},
      item: {
        display: 'inline-flex',
        alignItems: 'center',
        fontSize: '13px'
      },
      primary: {
        color:
          theme.palette.mode === 'light'
            ? theme.palette.primary.dark
            : theme.palette.primary.light
      },
      white: {
        color: theme.palette.common.white
      },
      yellow: {
        color:
          theme.palette.mode === 'light'
            ? theme.palette.warning.main
            : theme.palette.warning.light
      },
      icon: {
        display: 'inline-flex',
        marginRight: theme.spacing(1),
        fontSize: '13px'
      },
      text: {
        fontWeight: theme.typography.fontWeightBold
      },
      itemView: {
        marginBottom: theme.spacing(0.5),
        '& .MuiFlag-icon': {
          display: 'none'
        },
        '& $text': {
          backgroundColor: theme.palette.primary.dark,
          paddingLeft: theme.spacing(1),
          paddingRight: theme.spacing(1),
          height: theme.spacing(3),
          lineHeight: '24px',
          fontWeight: theme.typography.fontWeightRegular,
          position: 'relative',
          '&.shadow:after': {
            content: '""',
            width: 0,
            height: 0,
            filter: 'drop-shadow(1px 1px 1px rgba(0,0,0,.5))',
            borderColor:
              'rgba(0, 0, 0, 0.3) transparent transparent transparent',
            borderStyle: 'solid',
            borderWidth: '3px 2px 0 0',
            WebkitFilter: 'drop-shadow(1px 1px 1px rgba(0,0,0,.5))',
            position: 'absolute',
            right: 0,
            bottom: -3
          },
          '&$is_expires': {
            backgroundColor: theme.palette.error.main
          },
          '&$is_denied': {
            backgroundColor: theme.palette.error.main
          },
          '&$is_sponsor': {
            backgroundColor: theme.palette.warning.main
          },
          '&$is_pending': {
            backgroundColor: theme.palette.primary.light
          }
        }
      },
      detailView: {
        margin: theme.spacing(0.25, 0.5, 0),
        display: 'inline-flex',
        '& .MuiFlag-icon': {
          display: 'none'
        },
        '& $text': {
          backgroundColor: theme.palette.primary.dark,
          paddingLeft: theme.spacing(1),
          paddingRight: theme.spacing(1),
          height: theme.spacing(3),
          lineHeight: '24px',
          fontWeight: theme.typography.fontWeightRegular,
          position: 'relative',
          '&.shadow:after': {
            content: '""',
            width: 0,
            height: 0,
            filter: 'drop-shadow(1px 1px 1px rgba(0,0,0,.5))',
            borderColor:
              'rgba(0, 0, 0, 0.3) transparent transparent transparent',
            borderStyle: 'solid',
            borderWidth: '3px 2px 0 0',
            WebkitFilter: 'drop-shadow(1px 1px 1px rgba(0,0,0,.5))',
            position: 'absolute',
            right: 0,
            bottom: -3
          },
          '&$is_sponsor': {
            backgroundColor: theme.palette.warning.main
          },
          '&$is_pending': {
            backgroundColor: theme.palette.primary.light
          },
          '&$is_expires': {
            backgroundColor: theme.palette.error.main
          },
          '&$is_denied': {
            backgroundColor: theme.palette.error.main
          }
        }
      },
      is_sponsor: {},
      is_pending: {},
      is_expires: {},
      is_denied: {}
    }),
  {
    name: 'MuiFlag'
  }
);
