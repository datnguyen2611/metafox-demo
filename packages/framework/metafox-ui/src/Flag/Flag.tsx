import { useGlobal } from '@metafox/framework';
import clsx from 'clsx';
import React from 'react';
import LineIcon from '../LineIcon';
import useStyles from './Flag.styles';
import { styled } from '@mui/material';
import { ItemShape } from '../types';

export interface FlagProps {
  type?:
    | 'is_featured'
    | 'is_sponsor'
    | 'is_pending'
    | 'is_expires'
    | 'is_denied';
  color?: 'primary' | 'white' | 'yellow';
  variant?: 'feedView' | 'itemView' | 'detailView' | 'text';
  className?: string;
  'data-testid': string;
  value?: boolean;
  text?: string;
  showTitleMobile?: boolean;
  item?: ItemShape;
}

const TitleFlag = styled('span', {
  slot: 'TitleFlag',
  shouldForwardProp: props => props !== 'showTitleMobile'
})<{ showTitleMobile?: boolean }>(({ theme, showTitleMobile }) => ({
  [theme.breakpoints.down('sm')]: {
    ...(!showTitleMobile && {
      display: 'none'
    })
  }
}));

function Flag({
  type,
  color = 'primary',
  variant = 'feedView',
  'data-testid': testid,
  value,
  className,
  text,
  showTitleMobile
}: FlagProps) {
  const classes = useStyles();
  const { i18n } = useGlobal();

  if (!value) return null;

  let label = {
    icon: '',
    title: ''
  };
  switch (type) {
    case 'is_featured':
      label = {
        icon: 'ico-diamond',
        title: 'featured'
      };
      break;
    case 'is_sponsor':
      label = {
        icon: 'ico-sponsor',
        title: 'sponsored'
      };
      break;
    case 'is_pending':
      label = {
        icon: 'ico-clock-o',
        title: 'pending'
      };
      break;
    case 'is_expires':
      label = {
        icon: '',
        title: text
      };
      break;
    case 'is_denied':
      label = {
        icon: '',
        title: text
      };
      break;
    default:
  }

  return (
    <span className={clsx(classes.root, className, classes[variant])}>
      <span className={clsx(classes.item, classes[color])} data-testid={testid}>
        <span className={classes.icon}>
          <LineIcon icon={label.icon} />
        </span>
        <TitleFlag
          showTitleMobile={showTitleMobile}
          className={clsx(classes.text, classes[type])}
        >
          {i18n.formatMessage({ id: label.title })}
        </TitleFlag>
      </span>
    </span>
  );
}

export default Flag;
