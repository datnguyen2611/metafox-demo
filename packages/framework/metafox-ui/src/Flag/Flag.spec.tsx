import { GlobalProvider, Manager } from '@metafox/framework';
import { IntlProvider } from '@metafox/intl';
import { ThemeProvider } from '@metafox/layout';
import { render } from '@testing-library/react';
import * as React from 'react';
import Flag, { FlagProps } from './Flag';

const MockInfoLoader = () => <div data-testid="infoLoader" />;

describe('Flag', () => {
  const props: FlagProps = {
    'data-testid': 'featured',
    type: 'is_featured',
    color: 'white',
    variant: 'feedView'
  };
  const usePreference = jest.fn().mockImplementation(() => ({}));
  const getConfig = jest.fn().mockImplementation(() => {});
  const mockOn = jest.fn();
  const mockOff = jest.fn();

  const manager = Manager.factory({}).use({
    usePreference,
    useLoggedIn: () => false,
    eventCenter: { on: mockOn, off: mockOff },
    constants: { isLayoutPreviewWindow: false },
    getConfig
  });

  it('render', async () => {
    const { container, getByTestId } = render(
      <GlobalProvider value={manager}>
        <IntlProvider>
          <ThemeProvider infoLoader={MockInfoLoader}>
            <Flag value {...props} />
          </ThemeProvider>
        </IntlProvider>
      </GlobalProvider>
    );

    expect(getByTestId('featured')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('render color primary', async () => {
    const { container, getByTestId } = render(
      <GlobalProvider value={manager}>
        <IntlProvider>
          <ThemeProvider infoLoader={MockInfoLoader}>
            <Flag {...props} value color="primary" />
          </ThemeProvider>
        </IntlProvider>
      </GlobalProvider>
    );

    expect(getByTestId('featured')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('render null', async () => {
    const { container } = render(
      <GlobalProvider value={manager}>
        <IntlProvider>
          <ThemeProvider infoLoader={MockInfoLoader}>
            <Flag {...props} value={false} color="primary" />
          </ThemeProvider>
        </IntlProvider>
      </GlobalProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
