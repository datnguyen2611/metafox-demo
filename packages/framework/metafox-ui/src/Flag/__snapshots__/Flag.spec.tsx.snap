// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag render 1`] = `
<div>
  <div
    data-testid="infoLoader"
  />
  <span
    class="MuiFlag-root"
  >
    <span
      class="MuiFlag-item MuiFlag-white"
      data-testid="featured"
    >
      <span
        class="MuiFlag-icon"
      >
        <span
          class="ico ico-diamond ltr-1o2x2pn-MuiIcon-root"
        />
      </span>
      <span
        class="MuiFlag-text"
      >
        featured
      </span>
    </span>
  </span>
</div>
`;

exports[`Flag render color primary 1`] = `
<div>
  <div
    data-testid="infoLoader"
  />
  <span
    class="MuiFlag-root"
  >
    <span
      class="MuiFlag-item MuiFlag-primary"
      data-testid="featured"
    >
      <span
        class="MuiFlag-icon"
      >
        <span
          class="ico ico-diamond ltr-1o2x2pn-MuiIcon-root"
        />
      </span>
      <span
        class="MuiFlag-text"
      >
        featured
      </span>
    </span>
  </span>
</div>
`;

exports[`Flag render null 1`] = `
<div>
  <div
    data-testid="infoLoader"
  />
</div>
`;
